spa/src/utilities/addMiddleEllipsis.ts

```diff
@@ -0,0 +1,8 @@
+function addMiddleEllipsis(str: string) {
+  if (str.length > 35) {
+    return str.substring(0, 20) + '...' + str.substring(str.length - 10, str.length);
```

Comment:
```suggestion
    return str.substring(0, 20) + '…' + str.substring(str.length - 10, str.length);
```
e.g. a single Unicode char instead of three periods.